src/routes/transactions/transactions-view.service.ts

```diff
@@ -172,9 +172,17 @@ export class TransactionsViewService {
       chainId: args.chainId,
     });
 
-    const [{ fullAppData }, buyToken, sellToken] = await Promise.all([
-      // Decode hash of `appData`
-      this.swapsRepository.getFullAppData(args.chainId, twapStruct.appData),
+    // Decode hash of `appData`
+    const fullAppData = await this.swapsRepository.getFullAppData(
+      args.chainId,
+      twapStruct.appData,
+    );
```

Comment:
I've decupled this block from the tokens retrieval, so we can fail-fast if the app is not allowed.